internal/cli/up.go

```diff
@@ -245,20 +245,21 @@ func provideWebURL(webHost model.WebHost, webPort model.WebPort) (model.WebURL,
 }
 
 func provideAssetServer(mode model.WebMode, version model.WebVersion) (assets.Server, error) {
+	s, err := assets.NewEmbeddedServer()
 	if mode == model.ProdWebMode {
+		if err == nil {
+			return s, nil
+		}
 		return assets.NewProdServer(assets.ProdAssetBucket, version)
 	}
-	if mode == model.PrecompiledWebMode || mode == model.LocalWebMode {
-		path, err := web.StaticPath()
-		if err != nil {
-			return nil, err
-		}
-		pkgDir := assets.PackageDir(path)
-		if mode == model.PrecompiledWebMode {
-			return assets.NewPrecompiledServer(pkgDir), nil
-		} else {
-			return assets.NewDevServer(pkgDir, model.WebDevPort(webDevPort))
-		}
+	if mode == model.PrecompiledWebMode && err == nil {
```

Comment:
if you ask for precompiled mode, and there are no precompiled assets, seems like that should be an error